src/overlays/actors/ovl_En_Okuta/z_en_okuta.c

```diff
@@ -40,153 +54,1086 @@ ActorInit En_Okuta_InitVars = {
+
+    if (this->timer == 5) {
+        pos.x = this->actor.world.pos.x;
+        pos.y = this->actor.world.pos.y + 40.0f;
+        pos.z = this->actor.world.pos.z;
+
+        velocity.x = 0.0f;
+        velocity.y = -0.5f;
+        velocity.z = 0.0f;
+
+        EnOkuta_SpawnSmoke(&pos, &velocity, -20, play);
+        Actor_PlaySfx(&this->actor, NA_SE_EN_OCTAROCK_DEAD2);
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, 15.0f)) {
+        EnOkuta_SpawnSplash(this, play);
+        Actor_PlaySfx(&this->actor, NA_SE_EN_DAIOCTA_LAND);
+    }
+
+    if (this->timer < 3) {
+        Actor_SetScale(&this->actor, ((this->timer * 0.25f) + 1.0f) * 0.01f);
+        return;
+    }
+
+    if (this->timer < 6) {
+        Actor_SetScale(&this->actor, (1.5f - ((this->timer - 2) * 0.2333f)) * 0.01f);
+        return;
+    }
+
+    if (this->timer < 11) {
+        Actor_SetScale(&this->actor, (((this->timer - 5) * 0.04f) + 0.8f) * 0.01f);
+        return;
+    }
+
+    if (Math_StepToF(&this->actor.scale.x, 0.0f, 0.0005f)) {
+        SoundSource_PlaySfxAtFixedWorldPos(play, &this->actor.world.pos, 30, NA_SE_EN_COMMON_WATER_MID);
+        Item_DropCollectibleRandom(play, &this->actor, &this->actor.world.pos, 0xA0);
+
+        for (i = 0; i != 20; i++) {
+            velocity.x = (Rand_ZeroOne() - 0.5f) * 7.0f;
+            velocity.y = Rand_ZeroOne() * 7.0f;
+            velocity.z = (Rand_ZeroOne() - 0.5f) * 7.0f;
+            EffectSsDtBubble_SpawnCustomColor(play, &this->actor.world.pos, &velocity, &sBubbleAccel, &sBubblePrimColor,
+                                              &sBubbleEnvColor, Rand_S16Offset(100, 50), 25, false);
+        }
+
+        Actor_Kill(&this->actor);
+    }
+
+    this->actor.scale.z = this->actor.scale.x;
+    this->actor.scale.y = this->actor.scale.x;
+}
+
+void EnOkuta_SetupFrozen(EnOkuta* this, PlayState* play) {
+    this->timer = 10;
+    Actor_SetColorFilter(&this->actor, COLORFILTER_COLORFLAG_GRAY, COLORFILTER_INTENSITY_FLAG | 255,
+                         COLORFILTER_BUFFLAG_OPA, 10);
+    this->actor.child = Actor_SpawnAsChild(
+        &play->actorCtx, &this->actor, play, ACTOR_OBJ_ICEBLOCK, this->actor.world.pos.x,
+        this->actor.world.pos.y + this->skelAnime.jointTable->y * this->actor.scale.y + 25.0f * this->headScale.y,
+        this->actor.world.pos.z, 0, this->actor.home.rot.y, 0, 3);
+
+    if (this->actor.child != NULL) {
+        this->actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+        this->actor.flags |= ACTOR_FLAG_10;
+        this->actor.child->csId = this->actor.csId;
+        this->actionFunc = EnOkuta_FrozenInIceBlock;
+        return;
+    }
+
+    // The only way to reach this code is if our attempt to spawn the ObjIceblock actor failed for some reason. In the
+    // final game, this basically never happens during normal play, so the below code is effectively unused and never
+    // seen by most players, even if it *can* be run under certain extreme circumstances.
```

Comment:
I think the current explanation is better than just saying this code is unreachable

Reply:
But the code already speaks for itself aside from just the code appearing to be reachable due to it not being in a conditional code block.

With even the slightest bit of familiarity to the code base, it should be obvious to the reader that Actor_SpawnAsChild normally returns a non-null value (like so much so to the point where it's just assumed to always return an actor instance in some spots). It should also be obvious to the reader that Actor_SpawnAsChild has a rare potential failure case because we're checking if the return value is null.

We don't need a comment to explain that Actor_SpawnAsChild can fail in "extreme circumstances", because the code is already doing this for us. There's no important information to be gained from reading the comment that cannot just as easily be summed up by saying "This block is usually unreachable." I don't think we need a sentence to explain the meaning and consequences of unreachable code, which is what all of it is.

Reply:
Not everyone is familiar with the codebase, and not everyone who even looks at `Actor_SpawnAsChild` will be aware of its various failure methods. For example, this can fail if `OBJECT_ICE_BLOCK` isn't in the scene, which is something that is reasonably likely to come up in a modding context. However, I really have zero motivation to argue this further, so I just deleted the comment entirely.